### PR TITLE
CC | move image service related code into image-rpc.rs

### PR DESCRIPTION
**Pull request by jiangliu**
Move image service related code into image-rpc.rs, and also introduce a singleton instance of image service for coming refactor.

### CODE REVIEW

**Comment by ChengyuZhu6** [comment]
@jiangliu. I have rebuilt the tdx qemu test and the test is ok: http://10.112.240.228:8080/view/CCv0%20PR%20Triggered/job/kata-containers-CCv0-qemu-TDX-CRI_CONTAINERD_K8S-PR/1134/console

File: src/agent/src/image_rpc.rs [hunk 1]
```diff
@@ -240,10 +250,80 @@ impl ImageService {
...
+                    }
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    // Partially merge an OCI process specification into another one.
+    fn merge_oci_process(&self, target: &mut oci::Process, source: &oci::Process) {
```
Comment (stevenhorsman) [inline src/agent/src/image_rpc.rs:311]: Hey Gerry, as this code was moved from rpc.rs, can you add co-authors to this commit for the original authors, so we don't lose the DCO trail please. I've had a look and I think the original contributors were:
```
Co-authored-by: wllenyj <wllenyj@linux.alibaba.com>
Co-authored-by: jordan9500 <jordan.jackson@ibm.com>
Co-authored-by: stevenhorsman <steven@uk.ibm.com>
```
Thanks!

**Review by stevenhorsman — APPROVED** [review]
Small request to preserve original authors, but other than that this refactor LGTM. Thanks!

File: src/agent/src/image_rpc.rs (hunk 1 above)
Comment (jiangliu) [reply src/agent/src/image_rpc.rs:311]: sure:)

**Pull request merged**